### PR TITLE
Add "Using Strimzi with Local storage" blog post

### PR DESCRIPTION
**Pull request by scholzj**
This PR add the block post about using Kubernetes _Local_ storage with Kafka and Strimzi.

### CODE REVIEW

File: _posts/2018-06-11-using-strimzi-with-local-storage.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+It will start with an empty volume.
+It will therefore need to resync all its messages from one of the other replicas.
+Depending on the size of your partitions, this could take a very long time.
+Even if the original node was just restarted, the data stored in ephemeral volume usually do not survive the restart.
+The Pod with your broker will also usually not wait for the node to come back online and it will be started on another node.
+
+![Ephemeral storage during node crash]({{ "/assets/2018-06-11-using-strimzi-with-local-storage-ephemeral-storage.png" | absolute_url }})
+
+The choice between persistent network attached storage and ephemeral local storage is hard.
+But often - especially with large brokers storing terrabytes of data - the persistent network attached storage can be seen as the better option.
```
Comment (ppatierno) [inline _posts/2018-06-11-using-strimzi-with-local-storage.md]: typo "terabytes"

File: _posts/2018-06-11-using-strimzi-with-local-storage.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+So when the node restarts, the data which were stored on it survive.
+Additionally, since the Local volume is tied to a particular node, it will make sure that the Pod with your Kafka broker will not be rescheduled to another host.
+Kubernetes will wait until your node is back online and only then schedule the Kafka broker on it.
+When the broker is started, it will find the volume in the same state as it was left.
+It has to resync only the records which it missed during the failover - this is exactly the same as when you use persistent network attached storage.
+
+![Local persistent storage during node restart]({{ "/assets/2018-06-11-using-strimzi-with-local-storage-local-storage.png" | absolute_url }})
+
+Now ... what if your node is gone and will never return?
+In that case, you will need to make sure that the broker is scheduled on another node.
```
Comment (ppatierno) [inline _posts/2018-06-11-using-strimzi-with-local-storage.md]: How the user can achieve that? Should we explain here?

File: _posts/2018-06-11-using-strimzi-with-local-storage.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+Local storage is definitely an important improvement to Kubernetes.
+But is it really a good idea to use it for Kafka?
+
+When you are running your Kafka cluster on one of the three major public clouds providers, it might not be interesting.
+Regardless of whether you use Amazon AWS, Microsoft Azure or Google Cloud Platform, their network attached storage is able to deliver very good performance and their virtual machines are optimized for using it.
+On the other hand, machines with fast and large local storage are not that common and are not always cheaper than using network attached storage.
+Also the nodes of your Kubernetes cluster also tend to be treated differently in public cloud.
+Instead of doing regular maintenance and restarting the nodes, it is often easier to just spin a new fresh node and kill the old one.
+And as explained in previous chapters, this approach works perfectly with the networked attached storage.
```
Comment (ppatierno) [inline _posts/2018-06-11-using-strimzi-with-local-storage.md]: "networked" again ...

File: _posts/2018-06-11-using-strimzi-with-local-storage.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+But doesn't fit well with the Local storage, which will be lost during such operations.
+
+On the other hand, when you run Kafka on Kubernetes or OpenShift installed _on-premise_ in your own datacenter, the Local storage might be a real gamechanger.
+You will not need to invest in high performance storage network.
```
Comment (ppatierno) [inline _posts/2018-06-11-using-strimzi-with-local-storage.md]: "high performance network storage" ? or "high performance network attached storage" ?

File: _posts/2018-06-11-using-strimzi-with-local-storage.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+You just attach the storage locally and use it.
+If you run Kubernetes or OpenShift on bare-metal, your cluster nodes will be also treated as personal resource.
+They will be regularly restarted for maintenance reasons, but will usually get back online after the maintenance is finished.
+But the node will disappear only when it really dies or when it needs to be replaces due to age.
```
Comment (ppatierno) [inline _posts/2018-06-11-using-strimzi-with-local-storage.md]: typo "to be replaced"

File: _posts/2018-06-11-using-strimzi-with-local-storage.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
+---
+layout: post
+title:  "Using Strimzi with Local storage"
+date: 2018-06-11
+author: jakub_scholz
+---
+
+When deploying new Kafka cluster, one of the first decisions you have to make is the type of storage you want to use.
+Even before deciding on such storage aspects such as the type of filesystem which should be used, you have to decide a more basic question: should you run Kafka with local storage or with networked attached storage?
```
[the remark this answers is not in the document]
Comment (scholzj) [reply _posts/2018-06-11-using-strimzi-with-local-storage.md]: These are just typos. I don't know what makes me write _networked attached storage_ all the time ... but it is for sure not intentional :-o

File: _posts/2018-06-11-using-strimzi-with-local-storage.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+Kubernetes for a long time supported two kinds of storage:
+
+* Persistent network attached storage
+* Local ephemeral storage
+
+Each behaves a bit differently when your Pods with kafka brokers are rescheduled or when the node they are running on fails.
+The persistent network attached storage is not tied to any particular node.
+So when the node where your Kafka broker is running suddenly crashes, Kubernetes will be able to reallocate your broker to different node in your cluster.
+This usually takes between several seconds and several minutes, depending on your cloud provider and storage implementation.
+The persistent volume with the data used by the given broker will be moved together with the broker.
```
[the remark this answers is not in the document]
Comment (scholzj) [reply _posts/2018-06-11-using-strimzi-with-local-storage.md]: Well, basically the volume will be unmounted from the old node (unassigned) and mounted to the new node where the broker is rescheduled. I tried to rephrase it - please check if it is more understandable.

File: _posts/2018-06-11-using-strimzi-with-local-storage.md (hunk 2 above)
Comment (scholzj) [reply _posts/2018-06-11-using-strimzi-with-local-storage.md]: Basically, you delete the PVC and the Pod and wait until a new one is scheduled. Not sure whether we will come up with some more elegant solution. That is why I didn't went into the details. Maybe in next blog post :-)

File: _posts/2018-06-11-using-strimzi-with-local-storage.md (hunk 4 above)
Comment (scholzj) [reply _posts/2018-06-11-using-strimzi-with-local-storage.md]: This is intentional ... the network is usually the problem, not the NAS it self.

**Comment by scholzj** [comment]
BTW: How do you feel about the name **Using Strimzi with Local storage**? I was wondering is it shouldn't be more something like **Using Kafka with Local storage** ... because it actually talks only briefly about Strimzi. WDYT?

**Comment by ppatierno** [comment]
maybe "Deploying Kafka (on Kubernetes) with Local storage using Strimzi" ... so that we have Strimzi as well searching for local storage on Google :-) Btw not sure about the "on Kubernetes" in parenthesis.

**Comment by scholzj** [comment]
Yeah, I don't like the parenthesis ... but apart from that I like it. I will change it.

**Comment by ppatierno** [comment]
LGTM! Already approved! Great blog post! ;)

**Pull request merged**